gp-auto-extract.php

```diff
@@ -21,51 +23,125 @@
-		$this->source_type_templates = array( 
-											'none' 		=> '',
-											'github' 	=> 'https://github.com/%s/archive/master.zip',
-											'wordpress' => 'https://downloads.wordpress.org/plugin/%s.zip',
-											'custom' 	=> '%s',
-										);
+		$this->source_type_templates = array(
+			'none'      => '',
+			'github'    => 'https://github.com/%s/archive/%s.zip',
+			'wordpress' => 'https://downloads.wordpress.org/plugin/%s.zip',
+			'custom'    => '%s',
+		);
+		$this->url_credentials = array();
 
 		// Add the admin page to the WordPress settings menu.
 		add_action( 'admin_menu', array( $this, 'admin_menu' ), 10, 1 );
-		
+		add_action( 'admin_enqueue_scripts', array( $this, 'load_custom_wp_admin_style' ) );
+
 		// If the user has write permissions to the projects, add the auto extract option to the projects menu.
-		if( GP::$permission->user_can( wp_get_current_user(), 'write', 'project' ) ) {
-			add_action( 'gp_project_actions', array( $this, 'gp_project_actions'), 10, 2 );
+		if ( GP::$permission->user_can( wp_get_current_user(), 'write', 'project' ) ) {
+			add_action( 'gp_project_actions', array( $this, 'gp_project_actions' ), 10, 2 );
 		}
 
 		// We can't use the filter in the defaults route code because plugins don't load until after
 		// it has already run, so instead add the routes directly to the global GP_Router object.
-		GP::$router->add( "/auto-extract/(.+?)", array( $this, 'auto_extract' ), 'get' );
-		GP::$router->add( "/auto-extract/(.+?)", array( $this, 'auto_extract' ), 'post' );
+		GP::$router->add( '/auto-extract/(.+?)', array( $this, 'auto_extract' ), 'get' );
+		GP::$router->add( '/auto-extract/(.+?)', array( $this, 'auto_extract' ), 'post' );
+	}
+
+	/**
+	 * Enqueues the admin assets.
+	 *
+	 * @param String $hook Admin page hook name.
+	 */
+	public function load_custom_wp_admin_style( $hook ) {
+		// Load only on ?page=gp-auto-extract.php.
+		if ( 'settings_page_gp-auto-extract' !== $hook ) {
+			return;
+		}
+		wp_enqueue_style( 'gp-auto-extract-css', plugins_url( 'assets/css/gp-auto-extract.css', __FILE__ ) );
+
+		wp_register_script( 'gp-auto-extract-js', plugins_url( 'assets/js/gp-auto-extract.js', __FILE__ ) );
+		$translation_array = array(
+			'passwords' => array(
+				'none' => '',
+				'github' => esc_attr__( 'or Personal Access Token' ),
+				'wordpress' => '',
+				'custom' => '',
+			),
+			'settings' => array(
+				'none' => '',
+				'github' => esc_attr__( 'username/repository' ),
+				'wordpress' => esc_attr__( 'plugin-or-theme-slug' ),
+				'custom' => esc_attr__( 'url for a valid archive with source files' ),
+			),
+		);
+		wp_localize_script( 'gp-auto-extract-js', 'gpae', $translation_array );
+		wp_enqueue_script( 'gp-auto-extract-js' );
+	}
+
+	/**
+	 * This function is here as placeholder to support adding the auto extract option to the router.
+	 * Without this placeholder there is a fatal error generated.
+	 */
+	public function before_request() {
```

Comment:
This placeholder function was removed in the current tree, you must have added it back in by accident during the rebase.